src/prelude/internals.mo

```diff
@@ -565,6 +558,10 @@ func @timer_helper() : async () {
 
   gatherExpired(@timers);
 
+  let exp = @nextExpiration @timers;
+  ignore (prim "global_timer_set" : Nat64 -> Nat64) exp;
+  if (exp == 0) @timers := null;
+
   for (k in thunks.keys()) {
     ignore switch (thunks[k]) { case (?thunk) ?thunk(); case _ null };
   }
```

Comment:
```suggestion
  for (o in thunks.vals()) {
     switch o { 
       case (?thunk) { ignore thunk() } 
       case _ { } 
    };
  }
```
(or use `let-else` and `continue` from `else` if you are brave)

Reply:
Yes I was pondering that. But `let-else` has a slight performance penalty due to evaluation the pattern _twice_ (currently).